src/blockrelay/graphene_set.cpp

```diff
@@ -29,31 +30,41 @@ CGrapheneSet::CGrapheneSet(size_t _nReceiverUniverseItems,
 
     FastRandomContext insecure_rand(fDeterministic);
 
+    // Infer various receiver quantities
+    uint64_t nReceiverExcessItems =
+        std::max((int)(nReceiverUniverseItems - nItems), (int)(nSenderUniverseItems - nItems));
+    nReceiverExcessItems = std::max(0, (int)nReceiverExcessItems); // must be non-negative
+    nReceiverExcessItems =
+        std::min((int)nReceiverUniverseItems, (int)nReceiverExcessItems); // must not exceed total mempool size
+    uint64_t nReceiverMissingItems = std::max(1, (int)(nItems - (nReceiverUniverseItems - nReceiverExcessItems)));
+
+    LOG(GRAPHENE, "receiver expected to have at most %d excess txs in mempool\n", nReceiverExcessItems);
+    LOG(GRAPHENE, "receiver expected to be missing at most %d txs from block\n", nReceiverMissingItems);
+
     // Optimal symmetric differences between receiver and sender IBLTs
     // This is the parameter "a" from the graphene paper
-    double optSymDiff = 1;
+    double optSymDiff = std::max(1, (int)nReceiverMissingItems);
```

Comment:
redundant with the above std::max in nReceiverMissingItems creation.

Reply:
Good point. Fixed.